### ISSUE TITLE
Stepper: Localise `recordSignupStart` and ensure timers don't get restarted on rerenders

### ISSUE THREAD

**Issue by chriskmnds** [Automattic/wp-calypso#94767]
### What

As of https://github.com/Automattic/wp-calypso/pull/94743, we no longer limit the `calypso_signup_start` event to trigger only on initial load (e.g. when on first or User step) and will no longer limit the event to be triggered only once per session (so rerenders affecting the enclosing hook will record the same event again).

With the above, it makes sense to limit the timer used in [recordSignupStart](https://github.com/Automattic/wp-calypso/blob/trunk/client/lib/analytics/signup.js) (used for `elapsed_time_since_start`) to initialize only once per session. So only a page reload to reset the timer, not a rerender. A rerender could happen from `useEventProps` (or `useSignupStartEventProps`) fetching updated state values.

This is a follow-up issue from discussion in https://github.com/Automattic/wp-calypso/pull/94706. See:

>> Tested it a little and `elapsed_time_since_start` is null because `recordSignupStart` is called only the first time (and it sets the session storage with the starting time). You probably tested while already logged in and so the event is not triggered. [Stepper: Record `calypso_signup_start` unconditionally => multiple logs #94743](https://github.com/Automattic/wp-calypso/pull/94743) should solve it (trigger the events on every refresh)
>  
> Just to note here that with https://github.com/Automattic/wp-calypso/pull/94743 if a flow uses something like:
> 
> ```
> useEventProps() {
>   const { data } = useSiteIntent( 123 );
>   return { 
>     calypso_signup_start: {
>       site_intent: data?.site_intent,
>     } ,
>   };
> }
> ```
> 
> (or through the current `useSignupStartEventProps` - which is getting refactored in another PR) then `elapsed_time_since_start` will get reset when `recordSignupStart` gets triggered 😬 
> 
> I think there are two solutions (outside of ignoring `useEventProps` from dependencies in `useSignUpStartTracking`):
> 
> 1. Ignore/don't care - so let the flow be silly about it
> 2. Rework the `recordSignupStart` function to check if a previous value exists, so to not reset it if it exists.
> 
> I think it's not a blocker for the experiment as we don't use `extraProps` or `flowVariant` for onboarding flow. I can create an issue to follow up on this, if we proceed with https://github.com/Automattic/wp-calypso/pull/94743
> 
> _Originally posted by @chriskmnds in https://github.com/Automattic/wp-calypso/issues/94706#issuecomment-2363189857_             

### How

The current `recordSignupStart` utility is:

```
export function recordSignupStart( flow, ref, optionalProps ) {
	setSignupStartTime();

	// Tracks
	recordTracksEvent( 'calypso_signup_start', {
		flow,
		ref,
		...optionalProps,
	} );
	// Google Analytics
	gaRecordEvent( 'Signup', 'calypso_signup_start' );
	// Marketing
	adTrackSignupStart( flow );
}
```

One way to go about this is separate the Tracks event from the other trackers/timers in `useSignUpStartTracking`

1. Create `client/landing/stepper/declarative-flow/internals/analytics/record-signup-start.ts` and add the `// Tracks` block (like we have for the rest in that folder).
2. Move the remaining trackers into a `useEffect` call in `useSignUpStartTracking`:

```
useEffect( () => {
    setSignupStartTime();
    // Google Analytics
    gaRecordEvent( 'Signup', 'calypso_signup_start' );
    // Marketing
    adTrackSignupStart( flow );
}, [] );

```
### How to test

Use `/setup/ecommerce` that uses `useSignupStartEventProps` and will trigger the event multiple times when progressing through the steps.

cc @escapemanuele
